eng/Versions.props

```diff
@@ -103,20 +103,22 @@
-    <!-- Some tasks and the resolver will need to run in an MSBuild that is older than the very latest,
-          so target one that matches the version in minimumMSBuildVersion.
-
-          This avoids the need to juggle references to packages that have been updated in newer MSBuild. -->
-    <MicrosoftBuildPackageVersion Condition="exists('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion') and '$(DotNetBuildFromSource)' != 'true'">$([System.IO.File]::ReadAllText('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion').Trim())</MicrosoftBuildPackageVersion>
-    <MicrosoftBuildFrameworkPackageVersion>$(MicrosoftBuildPackageVersion)</MicrosoftBuildFrameworkPackageVersion>
-    <MicrosoftBuildRuntimePackageVersion>$(MicrosoftBuildPackageVersion)</MicrosoftBuildRuntimePackageVersion>
-    <MicrosoftBuildLocalizationPackageVersion>17.10.0-preview-24101-01</MicrosoftBuildLocalizationPackageVersion>
-    <MicrosoftBuildUtilitiesCorePackageVersion>$(MicrosoftBuildPackageVersion)</MicrosoftBuildUtilitiesCorePackageVersion>
-    <MicrosoftBuildTasksCorePackageVersion>$(MicrosoftBuildPackageVersion)</MicrosoftBuildTasksCorePackageVersion>
-    <MicrosoftBuildTasksCoreVersion>$(MicrosoftBuildTasksCorePackageVersion)</MicrosoftBuildTasksCoreVersion>
-    <MicrosoftNETStringToolsPackageVersion>$(MicrosoftBuildPackageVersion)</MicrosoftNETStringToolsPackageVersion>
+    <!-- In VMR modes, MicrosoftBuildVersion and MicrosoftBuildLocalizationVersion will end
+         up overwritten by properties flowing the newest version that are imported after this file.
+         These should be used in places where we always want to target the latest. This includes source-only modes (always use the latest)
+         as well as some additional cases like the msbuild that is redistributed in the SDK.
+        
+         In non-source-only modes, some tasks and the resolver will need to run in an MSBuild that is older
+         than the very latest, based on what we want the SDK to support. So use a version that matches the version
+         in minimumMSBuildVersion. In these cases, we don't want to use MicrosoftBuildVersion and other
+         associated properties that are updated by the VMR infrastructure. So, we read this version
+         from the 'minimumMSBuildVersion' file in non-source-only cases into MicrosoftBuildMinimumVersion,
+         then use that in Directory.Packages.props.
+         
+         At usage sites, either we use MicrosoftBuildMinimumVersion, or MicrosoftBuildVersion in source-only modes. -->
+    <MicrosoftBuildVersion>17.10.0-preview-24101-01</MicrosoftBuildVersion>
+    <MicrosoftBuildLocalizationVersion>17.10.0-preview-24101-01</MicrosoftBuildLocalizationVersion>
+    <MicrosoftBuildMinimumVersion Condition="exists('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion') and '$(DotNetBuildSourceOnly)' != 'true'">$([System.IO.File]::ReadAllText('$(RepoRoot)\src\Layout\redist\minimumMSBuildVersion').Trim())</MicrosoftBuildMinimumVersion>
```

Comment:
What happens if that file doesn't exist?

Reply:
MicrosoftBuildMinimumVersion will be empty and the build will fail spectacularly. We could set it to the MSBuildVersion but then we might miss if that file were removed during the build for some unforseen reason.